### ISSUE TITLE
Option to generate separate / unique maskable icon

### ISSUE THREAD

**Issue by cameronrr** [AllanChain/gridsome-plugin-pwa#28]
Ideally maskable icon can be specified separately to a regular icon.

> While you can specify multiple space-separated purposes like "any maskable", in practice you shouldn't. Using "maskable" icons as "any" icons is suboptimal as the icon is going to be used as-is, resulting in excess padding and making the core icon content smaller. Ideally, icons for the "any" purpose should have transparent regions and no extra padding, like your site's favicons, since the browser isn't going to add that for them.

Source: [Adaptive icon support in PWAs with maskable icons
](https://web.dev/maskable-icon/)

I recently submitted a pull request to gridsome-plugin-pwa to achieve this, I think it would be great to have this added here too. It looks like it can be handled similarly here with no impact on existing use, whereby the 'maskable' property of defaultIconConfig.androidChrome is expanded to allow for either a boolean or a string:

- When maskable is false, the icons are generated with purpose 'any' (current behaviour)
- When maskable is true, the icons are generated with purpose 'maskable any' (current behaviour)
- When maskable is string, it is treated as a unique icon, to be generated with purpose 'maskable' (new feature)

This gives developers the freedom to have both 'regular' icons and 'maskable' icons generated and identified in the manifest, per recommendations for adaptive icon support in PWAs.

**Comment by AllanChain** [comment]
How about using a list? I think that's more explicit.

```js
androidChrome: [
    {
      src: './src/icon.png',
      sizes: [512],
      purpose: 'any'
    },
    {
      src: './src/icon-maskable.png',
      sizes: [144, 96, 72, 48],
      purpose: 'maskable'
    },
    {
      urls: ['/icons/384x384.png', '/icons/192x192.png'],
      sizes: [384, 192],
      pupose: 'maskable any'
    }
  ]
```

To allow `maskable any`, `maskable`, and `any`, `purpose` is better than `maskable`.

**Comment by AllanChain** [comment]
Thanks for the idea and info! Implemented in `v0.4.0-alpha.1`

**Comment by cameronrr** [comment]
That's awesome thank you! Yes your approach is better, lets the developer have access to all the usual options for the unique icon then too, rather than mimicking the regular icon. I'll test it in my application using this package shortly, so far I like it's method similar to vue-cli and works well. Especially after adding the prompt to reload functionality.
